### PR TITLE
IBM Power S822LC machine support bmcsetup, build genesis based on Fedora 22 ppc64

### PR DESCRIPTION
**Pull request by zet809**
For building genesis with Fedora 22 ppc64 that include the latest ipmitool, pls following the steps below before build genesis-base package.
1. The Fedora 22 ppc64 can be installed either on pkvm VM or IBM Power S822LC machine.
2. Install ipmitool with yum install ipmitool. It will install the default ipmitool-1.8.13.
3. Download ipmitool-1.8.15 tar ball from https://sourceforge.net/projects/ipmitool/files/latest/download.
4. Install gcc
5. untar ipmitool-1.8.15 tar ball, cd into the top directory, then ./configure, make.
    The ipmitool-1.8.15 binary will be under ./src/.
6, replace /usr/bin/ipmitool with ./src/ipmitool
7. Then build xcat-genesis-base pkg with normal steps.

### CODE REVIEW

**Comment by whowutwut** [comment]
@zet809  can we add some information into the pull request? Should update title to:  "~~Firestone~~IBM Power S822LC" for customers to be able to search through the Pull Request/Issues

File: xCAT-genesis-scripts/bin/bmcsetup [hunk 1]
```diff
@@ -15,7 +15,12 @@
 allowcred.awk &
 CREDPID=$!
 sleep 5
-modprobe ipmi_si
+HOST_ARCH=`uname -m`
+if [ "$HOST_ARCH" = 'ppc64' ]; then
```
Comment (whowutwut) [inline xCAT-genesis-scripts/bin/bmcsetup]: does it matter ppc64le or ppc64?

File: xCAT-genesis-scripts/bin/bmcsetup [hunk 2]
```diff
@@ -374,22 +379,24 @@ if [ ! "$IPMIVER" == "1.5"  ]; then
...
-		echo "Not Needed"
-		break
-        fi
-        sleep 1
-        let TRIES=TRIES+1
-        if [ $TRIES -gt $TIMEOUT ]; then break; fi
-    done
-    if [ $TRIES -gt $TIMEOUT ]; then echo "ERROR"; else echo "OK"; fi
+    
+    # 0xc 0x21 with data 7 is used to set payload channel, it is readonly in the ipmiv2.0 doc, so comment this section out.
```
Comment (whowutwut) [inline xCAT-genesis-scripts/bin/bmcsetup]: I see you add comment on why commenting this section out. Is there a potential that we would need to bring it back into the code?

File: xCAT-genesis-scripts/bin/bmcsetup [hunk 3]
```diff
@@ -374,22 +379,24 @@ if [ ! "$IPMIVER" == "1.5"  ]; then
...
+    #    let TRIES=TRIES+1
+    #    if [ $TRIES -gt $TIMEOUT ]; then break; fi
+    #done
+    #if [ $TRIES -gt $TIMEOUT ]; then echo "ERROR"; else echo "OK"; fi
 fi
 
 # Reset the BMC for the x3755 M4 (8722), otherwise the BMC will not be pingable after running of bmcsetup
 XPROD=`ipmitool mc info|grep "^Product ID"|awk '{print $4}'`
-if [ "$XPROD" = "309" ] ; then
+if [ "$XPROD" = "309" -o "$XPROD" = "43707" ] ; then
```
Comment (whowutwut) [inline xCAT-genesis-scripts/bin/bmcsetup]: Can we add comment on what value 43707 represents?

File: xCAT-genesis-scripts/bin/bmcsetup (hunk 1 above)
Comment (zet809) [inline xCAT-genesis-scripts/bin/bmcsetup]: No, the genesis is build based on Fedora 22 ppc64, it won't return ppc64le when running.

File: xCAT-genesis-scripts/bin/bmcsetup (hunk 2 above)
Comment (zet809) [inline xCAT-genesis-scripts/bin/bmcsetup]: I don't think we will bring it back later. Maybe I need to delete this section.

File: xCAT-genesis-scripts/bin/bmcsetup (hunk 3 above)
Comment (zet809) [inline xCAT-genesis-scripts/bin/bmcsetup]: Yes, sure. 43707 is for Power S822LC and S812LC machine.

Comment (zet809) [inline xCAT-genesis-scripts/bin/bmcsetup]: I have updated the code with your comments, pls review.

**Comment by whowutwut** [comment]
New to this genesis build process..  The description just explains what you did... or are we going to ship a new version of genesis or just documenting these steps? 

I am okay to merge the changes

**Comment by zet809** [comment]
Our hardware discovery process is depended on xcat-genesis-base. 
So we must ship a new xcat-genesis-base ppc64 dependency pkg with 2.11.

**Comment by whowutwut** [comment]
@zet809  Ok, i am okay to merge if you are done with the changes.

**Comment by zet809** [comment]
Add comments in buildrpm to describe which OS is supported, and what else need to do before build xcat-genessis-base pkg. And checking 'ppc64*' to install ipmi_powernv module rather than 'ppc64'. 
Pls review.

**Comment by daniceexi** [comment]
I have reviewed the change and agree to merge.

**Comment by whowutwut** [comment]
Who is building this genesis package to add to the xcat-deps?

**Comment by daniceexi** [comment]
@zet809 Er Tao is working on the new build.

**Comment by zet809** [comment]
I have build xcat-genesis-base-2.11 based on Fedora 23 ppc64 rpm pkg, and upload it onto GSA. 
The xcat-dep both under linux and ubuntu have been added.

**Pull request merged**